bool refresh
---
int64 height
int64 width
int64 start
int64 end
int64 top
int64 bottom
int64 centerX
int64 centerY
bool flipped
